### PR TITLE
Implement WebSocket Security BBEs

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
$subject was done for following BBEs:

**WebSocket Security**
- Service - TLS
- Service - MTLS
- Service - Basic Auth File User Store
- Service - Basic Auth LDAP User Store
- Service - JWT Auth
- Service - OAuth2
- Client - TLS
- Client - MTLS
- Client Basic Auth
- Client Bearer Token
- Client - Self-signed JWT Auth
- Client - OAuth2 client credentials grant type
- Client - OAuth2 password
- Client - OAuth2 Refresh token

Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/1421
Related to https://github.com/ballerina-platform/ballerina-standard-library/issues/584

### CODE REVIEW

File: examples/websocket-client-basic-auth/websocket_client_basic_auth.bal [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        password: "ldclakmal@123"
+    },
+    secureSocket = {
+        cert: "../resource/path/to/public.crt"
+    }
+);
+
+public function main() returns error? {
+    check securedEP->writeTextMessage("Hello, World!");
+    string response = check securedEP->readTextMessage();
```
Comment (Bhashinee) [inline examples/websocket-client-basic-auth/websocket_client_basic_auth.bal:20]:
```suggestion
    string response = check securedEP->readTextMessage();
```
When we say the `response` isn't it more like getting a response for a request as in HTTP. Shall we keep it as  `textMessage`? WDYT?

Comment (ldclakmal) [reply examples/websocket-client-basic-auth/websocket_client_basic_auth.bal]: +1. Fixed with https://github.com/ballerina-platform/ballerina-distribution/pull/1930/commits/845ed035766a43de7340176f08ef9799c6df422f

File: examples/websocket-client-ssl-tls/websocket_client_ssl_tls.bal [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import ballerina/websocket;
+import ballerina/io;
+
+// An WebSocket client can be configured to communicate through WSS as well.
```
Comment (Bhashinee) [inline examples/websocket-client-ssl-tls/websocket_client_ssl_tls.bal:4]:
```suggestion
// A WebSocket client can be configured to communicate through WSS as well.
```

File: examples/websocket-client-self-signed-jwt-auth/websocket_client_self_signed_jwt_auth.out [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Before testing this sample, first start a sample service secured with JWT Auth.
+# (You may need to change the trusted certificate file path and private key file path.)
```
Comment (Bhashinee) [inline examples/websocket-client-self-signed-jwt-auth/websocket_client_self_signed_jwt_auth.out:2]:
```suggestion
# You may need to change the trusted certificate file path and private key file path.
```

File: examples/websocket-service-jwt-auth/websocket_service_jwt_auth.server.out [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# You may need to change the certificate file path, private key file path, and
+# trusted certificate file path.
```
Comment (Bhashinee) [inline examples/websocket-service-jwt-auth/websocket_service_jwt_auth.server.out:2]:
```suggestion
# You may need to change the certificate file path and private key file path.
```

File: examples/websocket-service-oauth2/websocket_service_oauth2.server.out [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# You may need to change the certificate file path, private key file path, and
+# trusted certificate file path.
```
Comment (Bhashinee) [inline examples/websocket-service-oauth2/websocket_service_oauth2.server.out:2]:
```suggestion
# You may need to change the certificate file path and private key file path.
```

**Review by Bhashinee — APPROVED** [review]

**Pull request merged**